service/src/lib.rs

```diff
@@ -150,7 +151,9 @@ pub fn run(
                             } else if auto_commit && predicate.can_commit {
                                 preprocessor.commit(predicate.texts[0].to_owned());
                             } else {
-                                frontend_tx1.send(GUICmd::Predicate(predicate))?;
+                                if suggestions {
+                                    frontend_tx1.send(GUICmd::Predicate(predicate))?;
+                                }
```

Comment:
@omer-biz 

The objective to make the autocompletion/autosuggestion on demand, is to speed up the typing using afrim.
Currently, you make the display on demand, but not the computation.

You should inhibit the translator.